src/core/search/search.service.ts

```diff
@@ -0,0 +1,403 @@
+    const profiles = await this.profileRepository.find({
+      relations: ['user', 'experience', 'skills'],
+      select: {
+        user: {
+          name: true,
+          lastname: true,
+          email: true,
+        },
+      },
+      where: {
+        deletedAt: null,
+      },
+    });
+
+    return profiles;
+  }
+
+  async createIndex() {
+    const checkIndex = await this.searchClient.indices.exists({
+      index: 'profiles',
+    });
+
+    if (checkIndex.statusCode === 404) {
+      const index = await this.searchClient.indices.create({
+        index: 'profiles',
+        body: {
+          settings: {
+            analysis: {
+              filter: {
+                autocomplete_filter: {
+                  type: 'edge_ngram',
+                  min_gram: 1,
+                  max_gram: 20,
+                },
+              },
+              analyzer: {
+                autocomplete: {
+                  type: 'custom',
+                  tokenizer: 'standard',
+                  filter: ['lowercase', 'autocomplete_filter'],
+                },
+              },
+            },
+          },
+          mappings: {
+            properties: {
+              id: { type: 'integer' },
+              userId: { type: 'integer' },
+              name: { type: 'text' },
+              lastname: { type: 'text' },
+              email: { type: 'text' },
+              description: { type: 'text' },
+              mainTitle: { type: 'text' },
+              countryResidence: { type: 'text' },
+              skills: {
+                type: 'nested',
+                properties: {
+                  name: { type: 'text' },
+                  type: { type: 'text' },
+                },
+              },
+              experience: {
+                type: 'nested',
+                properties: {
+                  businessName: { type: 'text' },
+                  role: { type: 'text' },
+                  location: { type: 'text' },
+                  description: { type: 'text' },
+                },
+              },
+            },
+          },
+        },
+      });
+    }
+  }
+
+  async indexProfiles() {
+    await this.createIndex();
+
+    const body = await this.parseAndPrepareData();
+
+    const resp = await this.searchClient.bulk({
+      index: 'profiles',
+      body,
+    });
+
+    return resp;
+  }
+
+  public async search(
+    searchParam: SearchProfileDto,
+    page: number,
+    limit: number,
+    random: number,
+    career: string[],
+    skills: string[],
+    countryResidence: string[],
+  ) {
+    try {
+      const from = (page - 1) * limit;
+
+      let filters = [];
+
+      if (!random) random = Math.floor(Math.random() * 1000);
+
+      career = await this.validateCarrera(career);
+
+      skills = await this.validateQueryArrayRelation(skills, 'skills');
+
+      countryResidence = await this.validateQueryArrayCountry(countryResidence);
+
+      if (career && Array.isArray(career)) {
```

Comment:
```suggestion
      if (career) {
```
Hace falta validar si es un arreglo? `validateCarrera` no lo hace ya?

Reply:
mmmmm creo que tienes razon pero deberia probar por algo lo puse y no recuerdo si era pq me lo devolvia como un string